pkg/gofr/datasource/solr/solr.go

```diff
@@ -168,7 +196,52 @@ type Response struct {
 }
 
 // call forms the http request and makes a call to solr and populates the solr response
-func call(ctx context.Context, method, url string, params map[string]any, body io.Reader) (any, error) {
+func (c *Client) call(ctx context.Context, method, url string, params map[string]any, body io.Reader) (any, error, trace.Span) {
+	var span trace.Span
+
+	if c.tracer != nil {
+		ctx, span = c.tracer.Start(ctx, fmt.Sprintf("Solr %s", method),
+			trace.WithAttributes(
+				attribute.String("solr.url", url),
+			),
+		)
+	}
+
+	req, err := c.createRequest(ctx, method, url, params, body)
```

Comment:
we can use `httptrace` package to trace the underlying requests and have the same context inherit the traceId from the parent call.

Reply:
sure done